### PR TITLE
Reply

### PR DESCRIPTION
**Pull request by jcorteza**

### CODE REVIEW

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java [hunk 1]
```diff
@@ -35,39 +35,27 @@ public TwitterService(Twitter twitterFactory) {
 
         logger.info("Attempting to update status through Twitter API.");
 
-        if (StringUtils.isEmpty(statusText)) {
-
-            logger.info("Twitter status update unsuccessful.");
+        verifyTextLength(statusText);
 
-            throw new TwitterServiceException(TwitterService.NO_TWEET_TEXT_MSG);
+        Optional<Status> responseOptional;
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitterapp/services/TwitterService.java]: create this when you need it

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java [hunk 2]
```diff
@@ -174,22 +191,36 @@ public Status createNewStatusObject(twitter4j.Status originalStatus) {
...
-                .append("https://twitter.com/")
-                .append(handle)
-                .append("/status/")
-                .append(originalStatus.getId())
-                .toString();
         newStatus.setMessage(originalStatus.getText());
         newStatus.setUser(newUser);
         newStatus.setCreatedAt(originalStatus.getCreatedAt());
-        newStatus.setPostUrl(url);
+        newStatus.setStatusID(originalStatus.getId());
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitterapp/services/TwitterService.java:209]: why the change to an existing api?

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java [hunk 3]
```diff
@@ -206,7 +237,7 @@ private void handleTwitterException(TwitterException e) throws TwitterServiceExc
 
         }
 
-        throw new TwitterServiceException("Twitter Exception thrown.", e);
+        return new TwitterServiceException("Twitter Exception thrown.", e);
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitterapp/services/TwitterService.java]: why is this returning?

File: src/test/java/com/khoros/twitterapp/MainResourceTest.java [hunk 4]
```diff
@@ -75,36 +77,36 @@ public void postStatusUpdateTestSuccess() {
...
         } catch (TwitterException e) {
 
-            Assert.fail("Test failed due to Twitter Exception.");
+            fail("Test failed due to Twitter Exception.");
 
         }
 
-        Assert.assertEquals(201, mainResource.postStatusUpdate(exampleText).getStatus());
-        Assert.assertEquals(twitterStatus.getText(), statusResponse.getMessage());
+        assertEquals(201, mainResource.postStatusUpdate(exampleText).getStatus());
```
Comment (kev-chan) [inline src/test/java/com/khoros/twitterapp/MainResourceTest.java]: use existing enums vs hardcoded error codes

File: src/test/java/com/khoros/twitterapp/MainResourceTest.java [hunk 5]
```diff
@@ -7,11 +7,13 @@
 import com.khoros.twitterapp.models.Status;
 import com.khoros.twitterapp.models.User;
 
+import static org.junit.Assert.assertEquals;
```
Comment (kev-chan) [inline src/test/java/com/khoros/twitterapp/MainResourceTest.java]: ensure tests are placed in proper directories

File: src/test/java/com/khoros/twitterapp/TwitterServiceTest.java [hunk 6]
```diff
@@ -122,11 +117,131 @@ public void getUserTimelineTestSuccess() {
...
+        }
+
+    }
+
+    @Test
+    public void verifyTextLengthTestTooLong() {
+
+        StringBuilder sb = new StringBuilder();
+
+        for(int i = 0; i < 10; i++) {
```
Comment (kev-chan) [inline src/test/java/com/khoros/twitterapp/TwitterServiceTest.java]: what is twitter expanded the length?

**Review by kev-chan — CHANGES_REQUESTED** [review]
ensure you've properly tested all cases

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java (hunk 1 above)
Comment (jcorteza) [reply src/main/java/com/khoros/twitterapp/services/TwitterService.java]: If I move this declaration the code won't work. This is where I need it. There's a try catch below this declaration. responseOptional is set within the try brackets.

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java (hunk 2 above)
Comment (jcorteza) [reply src/main/java/com/khoros/twitterapp/services/TwitterService.java:209]: I need to be able to access the status ID in order to set the inReplyTo for the reply. I would have had a statusID field from the beginning if I had known this functionality would be necessary later. Originally I was just trying to make it easier for the user of the api by supplying them with the tweet link, but the link can easily be built on their end as long as they have the statusID

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java (hunk 3 above)
Comment (jcorteza) [reply src/main/java/com/khoros/twitterapp/services/TwitterService.java]: I'm returning it to the original method throwing it there

File: src/main/java/com/khoros/twitterapp/TwitterApp.java [hunk 7]
```diff
@@ -22,5 +22,6 @@ public void run(TwitterAppConfiguration configuration, Environment environment)
                 .build();
 
         environment.jersey().register(resourceComponent.mainResource());
+        environment.jersey().register(SetDefaultHeaderFeature.class);
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitterapp/TwitterApp.java]: use the framework for the CORs

File: src/main/java/com/khoros/twitterapp/services/TwitterService.java [hunk 8]
```diff
@@ -20,6 +20,7 @@
 
     public static final int MAX_TWEET_LENGTH = 280;
     public static final String GENERAL_ERR_MSG = "Whoops! Something went wrong. Try again later.";
+    public static final String ID_ERR_MSG = "No value entered for inReplyTo.";
```
Comment (kev-chan) [inline src/main/java/com/khoros/twitterapp/services/TwitterService.java]: better variable name

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/test/java/com/khoros/twitterapp/services/TwitterServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,357 @@
...
+        }
+
+        assertEquals(exampleStatus.getText(), response.get().getMessage());
+        assertEquals(exampleStatus.getCreatedAt(), response.get().getCreatedAt());
+        assertEquals(exampleStatus.getId(), response.get().getStatusID());
+        assertEquals(testUrl, response.get().getPostUrl());
+
+    }
+
+//    @Test
```
Comment (kev-chan) [inline src/test/java/com/khoros/twitterapp/services/TwitterServiceTest.java]: remove if you don't need it

**Review by kev-chan — CHANGES_REQUESTED** [review]
